share/eds/utils.go

```diff
@@ -0,0 +1,9 @@
+package eds
+
+import "io"
+
+func closeAndLog(name string, closer io.Closer) {
+	if err := closer.Close(); err != nil {
+		log.Warnw("closing "+name, "err", err)
```

Comment:
will this not spam logs?

Reply:
If close failed on accessor, it means this accessor is leaked and will be never closed. We would want to know about such cases to hunt them down and handle properly. So this log should not spam, unless something is broken

Reply:
I don't know why I thought that it was logging on every close and not just on errors - I must have misread something. agree with you